collect_app/src/main/java/org/odk/collect/android/activities/FormDownloadList.java

```diff
@@ -339,8 +339,9 @@ private void downloadFormList() {
     @Override
     protected void onRestoreInstanceState(Bundle state) {
         super.onRestoreInstanceState(state);
-        toggleButtonLabel(toggleButton, listView);
-        updateAdapter();
+        if (PermissionUtils.areStoragePermissionsGranted(this)) {
+            updateAdapter();
```

Comment:
I can't recall why it's needed here, maybe it's not at all but I guess there is a tricky case that needs this fix so I decided to keep it and add if statement instead of removing what also would fix the crash but could bring some strange problems.